client/API.md

```diff
@@ -0,0 +1,536 @@
+| duration  | Duration string indicating how long to record the stream.           |
+
+##### Batch
+
+| Parameter | Purpose                                                                                                          |
+| --------- | -------                                                                                                          |
+| name      | Name of a task, records the results of the queries defined in the task.                                          |
+| start     | Earliest date for which data will be recorded. RFC3339Nano formatted.                                            |
+| stop      | Latest date for which data will be recorded. If not specified uses the current time. RFC3339Nano formatted data. |
+| past      | Duration string indicating how far into the past to start recording data relative to the stop date.              |
+| cluster   | Name of a configured InfluxDB cluster. If empty uses the default cluster.                                    |
+
+Only one of `start` or `past` may be specified.
+
+##### Query
+
+| Parameter | Purpose                                                                       |
+| --------- | -------                                                                       |
+| type      | Type of recording, `stream` or `batch`.                                   |
+| query     | Query to execute.                                                         |
+| cluster   | Name of a configured InfluxDB cluster. If empty uses the default cluster. |
+
+>NOTE: A recording itself is typed as either a stream or batch recording and can only be replayed to a task of a corresponding type.
+Therefore when you record the result of a raw query you must specify the type recording you wish to create.
+
+
+#### Example
+
+Create a recording using the `stream` method
+
+```
+POST /recording
+{
+    "method" : "stream",
+    "name" : "TASK_NAME",
+    "duration" : "10m"
+}
+```
+
+
+Create a recording using the `batch` method specifying a start time.
+
+```
+POST /recording
+{
+    "method" : "batch",
+    "name" : "TASK_NAME",
+    "start" : "2006-01-02T15:04:05Z07:00"
+}
+```
+
+Create a recording using the `batch` method specifying a past duration.
+
+```
+POST /recording
+{
+    "method" : "batch",
+    "name" : "TASK_NAME",
+    "past" : "7d"
+}
+```
+
+Create a recording using the `query` method specifying a `stream` type.
+
+```
+POST /recording
+{
+    "method" : "query",
+    "query" : "SELECT mean(usage_idle) FROM cpu WHERE time > now() - 1h GROUP BY time(10m)",
+    "type" : "stream"
+}
+```
+
+Create a recording using the `query` method specifying a `batch` type.
+
+```
+POST /recording
+{
+    "method" : "query",
+    "query" : "SELECT mean(usage_idle) FROM cpu WHERE time > now() - 1h GROUP BY time(10m)",
+    "type" : "batch"
+}
+```
+
+
+#### Response
+
+All recordings are assigned an ID which is returned in this format.
+
+```
+{
+    "id" : "e24db07d-1646-4bb3-a445-828f5049bea0"
```

Comment:
I think this should respond with a Location of the nearly created resource:

```http
HTTP 201 Created
Location: http://kapa.com/recording/e24db07d-1646-4bb3-a445-828f5049bea0
```

and/or a link in the response:

```json
{
    "id" : "e24db07d-1646-4bb3-a445-828f5049bea0",
    "link": {"rel": "self", "href": "/recording/e24db07d-1646-4bb3-a445-828f5049bea0"}
}
```